latch_cli/centromere/ctx.py

```diff
@@ -83,51 +86,64 @@ def __init__(
             else:
                 self.account_id = ws
 
-            self.pkg_root = Path(pkg_root).resolve()
             self.dkr_repo = config.dkr_repo
             self.remote = remote
-            self.container_map = {}
-
-            default_dockerfile = self.pkg_root.joinpath("Dockerfile")
-            if not default_dockerfile.exists():
-                generate_dockerfile(
-                    self.pkg_root, self.pkg_root.joinpath(".latch/Dockerfile")
-                )
-                default_dockerfile = self.pkg_root.joinpath(".latch/Dockerfile")
+            self.disable_auto_version = disable_auto_version
+            self.pkg_root = Path(pkg_root).resolve()
 
-            _import_flyte_objects([self.pkg_root])
+            # TODO (kenny) better rules to auto detect workflow type
+            if (
+                self.pkg_root.joinpath("Snakefile").exists()
```

Comment:
glob for Snakefile and `*.smk`?  Consider that `pkg_root` should just be the snakefile to run, since one may have repos with multiple snakemake entrypoints (i.e. multiple top-level workflows).

Might also not be a bad idea to have an opt-in CLI option to explicitly tell the latch cli the type

Reply:
re: entrypoint -  certainly quite crude as is, unfamiliar with the different ways of structuring snakemake projects + hope to tease this out with more test examples. will start with what you've provided